.github/workflows/merlin.yml

```diff
@@ -147,7 +147,7 @@ jobs:
         uses: golangci/golangci-lint-action@v3
         with:
           # Ensure the same version as the one defined in Makefile
-          version: v1.51.2
+          version: v1.56.2
```

Comment:
We need to update this since using an older version of the `golanci-lint` linter for yet to be supported Go versions potentially causes OOM [issues](https://github.com/golangci/golangci-lint/issues/3538) (the thread doesn't mention about Go 1.22 but it seems like the same issue happens nonetheless). `v1.56.0` is the earliest version that [supports](https://github.com/golangci/golangci-lint/releases/tag/v1.56.0) Go 1.22 but I just use the latest one here.